File: 1-Introductory_Problems/6-Number_Spiral/solution.hs
import Control.Monad (forM_)

main :: IO ()
main = do
  n <- read <$> getLine :: IO Integer
  forM_ [1..n] solve
    where solve _ = do
            [row, col] <- map read . words <$> getLine :: IO [Integer]
            print $ let layer = max row col in
                      if even layer
                        then layer * layer - (layer - row) - (col - 1)
                        else layer * layer - (layer - col) - (row - 1)

File: 1-Introductory_Problems/6-Number_Spiral/solution.cpp
#include <iostream>

int	main(void)
{
	int			n;
	long long	col;
	long long	row;
	long long	layer;
	long long	res;

	std::cin >> n;
	while (n--)
	{
		std::cin >> row >> col;
		layer = std::max(row, col);
		res = layer * layer;
		if (layer % 2)
			res -= (layer - col) + (row - 1);
		else
			res -= (layer - row) + (col - 1);
		std::cout << res << std::endl;
	}
}
